src/core/Group.cpp

```diff
@@ -305,8 +314,8 @@ void Group::setExpanded(bool expanded)
 {
     if (m_data.isExpanded != expanded) {
         m_data.isExpanded = expanded;
-        updateTimeinfo();
         if (config()->get("IgnoreGroupExpansion").toBool()) {
+            updateTimeinfo();
```

Comment:
@ckieschnick I'm not even sure we should be updating the time info in that case. @phoerious thoughts?

Reply:
I would say that depends on what KeePass does.

Reply:
Not quite sure if it is required by KeePass - the change was necessary due to retain the behavior after changing the timestamp update to use the self-listen mechanism with modified-signal (which is explicitly not emitted when "IgnoreGroupExpansion" is on). I think, if the behavior deviates from KeePass, it should be fixed separately.